### PR TITLE
fix zoom updated

### PR DESCRIPTION
**Pull request by SabrinaTardio**
Task/Issue URL: https://app.asana.com/0/1207686573307971/1208100557904771/f

**Description**: Fix an issue where a cycle of same document navigations are triggered.
This is fixed by updating the zoom on navigationDidCommit rather than on tab content change.
A second step to avoid to reload shield animation on the same page is webViewDidFinishNavigationPublisher updates only when not same-document navigations.

**Steps to test this PR**:
1. go to https://www.realtor.com/realestateandhomes-search/Duluth_MN?view=map&pos=46.955704,-92.397107,46.584657,-91.833761,10.881999999999996&qdm=true and check that the dashboard is not flickering and zooming the page and zooming or viewing the maps works as expected
2. Go to a website zoom in or out then in a new tab visit the same website and check the zoom is at the previously setting value immediately (it’s not adjusted on the go)
3. cmd+click on a link and check that the opened tab with the website has the same zoom
4. Set the zoom on bbc.com
5. go to bbc.co.uk which should reload to bbc.com check that after the reload the zoom is set as expected

<!—
Tagging instructions
If this PR isn't ready to be merged for whatever reason it should be marked with the `DO NOT MERGE` label (particularly if it's a draft)
If it's pending Product Review/PFR, please add the `Pending Product Review` label.

If at any point it isn't actively being worked on/ready for review/otherwise moving forward (besides the above PR/PFR exception) strongly consider closing it (or not opening it in the first place). If you decide not to close it, make sure it's labelled to make it clear the PRs state and comment with more information.
—>

**Definition of Done**:

* [ ] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

—
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)

### CODE REVIEW

File: DuckDuckGo/Tab/Model/Tab.swift [hunk 1]
```diff
@@ -377,7 +377,22 @@ protocol NewWindowPolicyDecisionMaker {
     }
 
     var webViewDidFinishNavigationPublisher: some Publisher<Void, Never> {
```
Comment (SabrinaTardio) [inline DuckDuckGo/Tab/Model/Tab.swift:379]: This is only used for that use case so it should be less risky then filter on the state publisher

In any case if you want we can do this later on it’s in the last commit that I can revert… the other changes still greatly improve the situation.

File: DuckDuckGo/Tab/ViewModel/TabViewModel.swift [hunk 2]
```diff
@@ -178,9 +178,14 @@ final class TabViewModel {
             .switchToLatest()
             .sink { [weak self] _ in
                 guard let self else { return }
-
                 updateAddressBarStrings()
                 updateFavicon()
+            }
+            .store(in: &cancellables)
+
+        tab.$hasCommittedContent
```
Comment (mallexxx) [inline DuckDuckGo/Tab/ViewModel/TabViewModel.swift:186]: Please add a comment explaining why this publisher is used here

**Review by mallexxx — APPROVED** [review]
The fix works good!
Although I‘ve noticed a couple of non-related issues:
- When zooming using a pinch gesture the zoom level is not saved between the navigation and can‘t be restored using Cmd+/-/0 hotkeys
- Trackers animation is not played on tripadvisor.com as the tracker detection script sends its updates after the navigation finishes but the animation trigger is happening on `didFinish`, causing the animation to go missing. I‘ve targeted it in https://github.com/duckduckgo/macos-browser/pull/3150
- Please add a comment to the added parts as it may be hard to follow for somebody not in the context
